monai/apps/pathology/transforms/spatial/array.py

```diff
@@ -73,3 +75,152 @@ def get_params(self, image_size):
         )
 
         return patch_size, steps
+
+
+class TileOnGrid(Randomizable, Transform):
+    """
+    Tile the 2D image into patches on a grid and maintain a subset of it.
+    This transform works only with np.ndarray inputs for 2D images.
+
+    Args:
+        tile_count: number of tiles to extract, if None extracts all non-background tiles
+            Defaults to ``None``.
+        tile_size: size of the square tile
+            Defaults to ``256``.
+        step: step size
+            Defaults to ``None`` (same as tile_size)
+        random_offset: Randomize position of the grid, instead of starting from the top-left corner
+            Defaults to ``False``.
+        pad_full: pad image to the size evenly divisible by tile_size
+            Defaults to ``False``.
+        background_val: the background constant (e.g. 255 for white background)
+            Defaults to ``255``.
+        filter_mode: mode must be in ["min", "max", "random", None]. If total number of tiles is more than tile_size,
+            then sort by intensity sum, and take the smallest (for min), largest (for max) or random (for random or None) subset
+            Defaults to ``min`` (which assumes background is high value)
+
+    """
+
+    def __init__(
+        self,
+        tile_count: Optional[int] = None,
+        tile_size: int = 256,
+        step: Optional[int] = None,
+        random_offset: bool = False,
+        pad_full: bool = False,
+        background_val: int = 255,
+        filter_mode: Optional[str] = "min",
+    ):
+        self.tile_count = tile_count
+        self.tile_size = tile_size
+        self.step = step
+        self.random_offset = random_offset
+        self.pad_full = pad_full
+        self.background_val = background_val
+        self.filter_mode = filter_mode
+
+        if self.step is None:
+            self.step = self.tile_size  # non-overlapping grid
+
+        self.offset = (0, 0)
+        self.random_idxs = np.array((0,))
+
+    def randomize(self, img_size: Sequence[int]) -> None:
+
+        c, h, w = img_size
+        tile_step = cast(int, self.step)
```

Comment:
There is no need to cast to `int`. If you correct its assignment in `__init__`, `self.step` has to be int.

Reply:
self.step is Optional(int),  if default None is provided it is always set to int in __init__ function. 
Nevertheless, without cast() function, mypy complains on all consecutive operations with tile_stem (such as % /, etc)
I can't pass the mypy test without it

Reply:
how would you do it without casting?

Reply:
without casting I till get, these errors (let's keep casting for mypy, it actually doesn't do anything it just gives mypy a hint)

`monai/apps/pathology/transforms/spatial/array.py:155:15: error: Unsupported operand types for // ("int" and "None") [operator] monai/apps/pathology/transforms/spatial/array.py:155:15: note: Right operand is of type "Optional[int]" monai/apps/pathology/transforms/spatial/array.py:155:35: error: Unsupported operand types for + ("int" and "None") [operator] monai/apps/pathology/transforms/spatial/array.py:156:15: error: Unsupported operand types for // ("int" and "None") [operator] monai/apps/pathology/transforms/spatial/array.py:156:15: note: Right operand is of type "Optional[int]" monai/apps/pathology/transforms/spatial/array.py:156:35: error: Unsupported operand types for + ("int" and "None") [operator]`

Reply:
If it has done correctly, we shouldn't need casting but let's keep it as is and I will fix it in a followup PR.